daemon/oci_opts.go

```diff
@@ -0,0 +1,23 @@
+package daemon
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/containers"
+	coci "github.com/containerd/containerd/oci"
+	"github.com/docker/docker/container"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// WithConsoleSize sets the initial console size
+func WithConsoleSize(c *container.Container) coci.SpecOpts {
+	return func(ctx context.Context, _ coci.Client, _ *containers.Container, s *coci.Spec) error {
+		if c.HostConfig.ConsoleSize[0] > 0 || c.HostConfig.ConsoleSize[1] > 0 {
```

Comment:
maybe add a fool-guard check that `len(HostConfig.ConsoleSize) == 2` here
(That's a bit sad we don't have a well defined struct here)

Reply:
Is it really needed given that ConsoleSize has type [2]uint? One could assume that it's len will always be 2, but maybe it's some Golang's gotcha that I'm not aware of?

Reply:
Double-checked (as arrays are not very commonly used in Go (nor recommended for most uses)), but yeah `[2]int` is a type, so always has 2 as len; https://go.dev/play/p/UxzMidJXmnb

trying to assign a `[]int` slice, even with len(2) will fail;

```go
var foo [2]int
foo = []int{1, 2}
```

That will produce:

```
cannot use []int{…} (value of type []int) as type [2]int in assignment
```


Good "catch" though! (definitely not a useless comment @ndeloof 👍 )